### PR TITLE
Added - Text can be copy from terminal 

### PR DESCRIPTION
**Pull request by milindk8**
# Description

Allow text to be copied from terminal UI - add feature to copy text from terminal

Fixes # https://github.com/devtron-labs/devtron/issues/849

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
All changes have been tested manually.

- [x] By copying text from terminal manually in sh/bash/powershell/cmd mode.
- [x] Pasting and checking manually the copied text is whether same or not. 

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

**Review by vivek-devtron — APPROVED** [review]

**Review by vivek-devtron — APPROVED** [review]

File: src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetailTabs/terminal/Terminal.tsx [hunk 1]
```diff
@@ -331,6 +347,33 @@ function TerminalView(terminalViewProps: TerminalViewProps) {
...
+            <AutoSizer>
+                {({ height, width }) => (
+                    <ResizableLogs showCopyToast={popupText} fitAddon={fitAddon} height={height} width={width} />
+                )}
+            </AutoSizer>
+        </div>
+    );
+}
+
+function ResizableLogs({ fitAddon, height, width, showCopyToast }) {
```
Comment (shivani170) [inline src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetailTabs/terminal/Terminal.tsx:359]: this function is repeating in both `LogViewer.component.tsx` as well in `Terminal.tsx`

File: src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetailTabs/terminal/Terminal.tsx [hunk 2]
```diff
@@ -331,6 +347,33 @@ function TerminalView(terminalViewProps: TerminalViewProps) {
                     {terminalViewProps.socketConnection}
                 </p>
             )}
+            <AutoSizer>
```
Comment (shivani170) [inline src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetailTabs/terminal/Terminal.tsx:350]: this function is repeating in both LogViewer.component.tsx as well in Terminal.tsx

File: src/components/LogViewer/LogViewer.tsx [hunk 3]
```diff
@@ -122,8 +121,9 @@ const LogViewer: React.FunctionComponent<logViewerInterface> = ({ subject, rootC
     return (
         <>
         <AutoSizer>
```
Comment (vivek-devtron) [inline src/components/LogViewer/LogViewer.tsx:123]: Remove AutoSizer, it is not required.

File: src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetailTabs/ResizableLogs.tsx [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+import  './nodeDetailTab.scss';
+
+interface Resize {
+    showCopyToast: boolean;
+    height:number;
+    width:number;
+}
+
+function ResizableLogs({ showCopyToast ,height,width}: Resize) {
+    return <Resizable />;
```
Comment (shivani170) [inline src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetailTabs/ResizableLogs.tsx:14]: In this case You can directly return div rather using function

File: src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetailTabs/terminal/Terminal.tsx [hunk 5]
```diff
@@ -283,6 +291,13 @@ function TerminalView(terminalViewProps: TerminalViewProps) {
             });
     };
 
+    function handleSelectionChange() {
```
Comment (shivani170) [inline src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetailTabs/terminal/Terminal.tsx:294]: repeating function. Make it DRY

File: src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetailTabs/ResizableLogs.tsx (hunk 4 above)
Comment (shivani170) [reply src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetailTabs/ResizableLogs.tsx:14]: Move Repetitive component in the common folder

File: src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetailTabs/CopyToast.tsx [hunk 6]
```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import { copyToClipboard } from '../../../../../common';
+import { ReactComponent as CheckIcon } from '../../../../assets/icons/ic-check.svg';
+import {AutoSizer} from 'react-virtualized';
+import '../../../../../../../node_modules/xterm/css/xterm.css';
+import '../../../../../LogViewer/LogViewer.scss';
+import  './nodeDetailTab.scss';
+
+
+function CopyToast({ showCopyToast }) {
```
Comment (shivani170) [inline src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetailTabs/CopyToast.tsx:10]: type is missing

**Review by shivani170 — APPROVED** [review]

**Review by vivek-devtron — APPROVED** [review]

**Pull request merged**